Source/ACE/Entity/Player.cs

```diff
@@ -1076,47 +1076,50 @@ public void AddFriend(string friendName)
                 newFriend.Id = new ObjectGuid(friendInfo.Guid, GuidType.Player);
 
                 // Save to DB, assume success
-                DatabaseManager.Shard.AddFriend(Guid.Low, newFriend.Id.Low);
-
-                // Add to character object
-                Character.AddFriend(newFriend);
+                DatabaseManager.Shard.AddFriend(Guid.Low, newFriend.Id.Low, (() =>
+                {
+                    // Add to character object
+                    Character.AddFriend(newFriend);
```

Comment:
This was already in the Task handler for GetObjectInfoByName.  This is not an issue introduced by this PR.